worlds/alttp/Bosses.py

```diff
@@ -224,20 +240,20 @@ def can_place_boss(boss: str, dungeon_name: str, level: Optional[str] = None) ->
                                                for boss in boss_table if not boss.startswith("Agahnim"))
 
 
-def place_boss(world, player: int, boss: str, location: str, level: Optional[str]) -> None:
-    if location == 'Ganons Tower' and world.mode[player] == 'inverted':
+def place_boss(multiworld, world, player: int, boss: str, location: str, level: Optional[str]) -> None:
+    if location == 'Ganons Tower' and multiworld.mode[player] == 'inverted':
         location = 'Inverted Ganons Tower'
     logging.debug('Placing boss %s at %s', boss, location + (' (' + level + ')' if level else ''))
-    world.get_dungeon(location, player).bosses[level] = BossFactory(boss, player)
+    world.dungeons[location].bosses[level] = BossFactory(boss, player)
 
 
 def format_boss_location(location: str, level: str) -> str:
     return location + (' (' + level + ')' if level else '')
 
 
-def place_bosses(world, player: int) -> None:
+def place_bosses(multiworld, world, player: int) -> None:
     # will either be an int or a lower case string with ';' between options
-    boss_shuffle: Union[str, int] = world.boss_shuffle[player].value
+    boss_shuffle: Union[str, int] = multiworld.boss_shuffle[player].value
```

Comment:
same as above and also the random might go away when/if we fade out non-slot-random? multiworld is only required for random and the one option, so you could capture both from world.multiworld at the top.

Reply:
this should be fixed now